src/utils/getFinalProducts.jsx

```diff
@@ -0,0 +1,72 @@
+import { useFilter } from "../contexts/filter-context";
+import { useProducts } from "../hooks";
+
+const getFinalProducts = () => {
+  const { products } = useProducts();
+
+  const { filterState } = useFilter();
+
+  const getCategoryProducts = (state, products) => {
+    return products.filter((prod) =>
+      state.categoryName.some((cat) => prod.categoryName.includes(cat))
+    );
+  };
+  
+  const getSortedProducts = (state, products) => {
+    switch (state.sorting) {
+      case "lth":
+        return products.sort(
+          (a, b) =>
+            a.price -
+            (a.price * a.discount) / 100 -
+            (b.price - (b.price * b.discount) / 100)
```

Comment:
Better to convert to Number type to ensure no side effects

Reply:
Made suggested changes to the code. Thank you for reviewing.